tests/src/main/java/alluxio/master/backcompat/BackwardsCompatibilityJournalGenerator.java

```diff
@@ -0,0 +1,132 @@
+import alluxio.Constants;
+import alluxio.ProjectConstants;
+import alluxio.master.backcompat.ops.AsyncPersist;
+import alluxio.master.backcompat.ops.CreateDirectory;
+import alluxio.master.backcompat.ops.CreateFile;
+import alluxio.master.backcompat.ops.DeleteFile;
+import alluxio.master.backcompat.ops.Mount;
+import alluxio.master.backcompat.ops.PersistDirectory;
+import alluxio.master.backcompat.ops.PersistFile;
+import alluxio.master.backcompat.ops.RenameFile;
+import alluxio.master.backcompat.ops.SetAcl;
+import alluxio.multi.process.MultiProcessCluster;
+import alluxio.multi.process.PortCoordination;
+import alluxio.security.LoginUser;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Generates journals for consumption by the BackwardsCompatibilityIntegrationTest.
+ *
+ * This class
+ * 1. starts a new cluster
+ * 2. runs all the test operations listed in OPS
+ * 3. takes a journal backup at stores it to src/test/resources/old_journals
+ * 4. stops the cluster
+ * 5. copies the journal to src/test/resources/old_journals
+ *
+ * Later, BackwardsCompatibilityIntegrationTest will start clusters from the backups and copies
+ * in old_journals, and validate that the operations' checks pass.
+ */
+public final class BackwardsCompatibilityJournalGenerator {
+  // Path relative to tests/src/test
+  public static final String OLD_JOURNALS_RESOURCE = "src/test/resources/old_journals";
+
+  public static final List<TestOp> OPS = ImmutableList.<TestOp>builder()
+      .add(new CreateDirectory(),
+          new CreateFile(),
+          new Mount(),
+          new AsyncPersist(),
+          new DeleteFile(),
+          new PersistFile(),
+          new PersistDirectory(),
+          new RenameFile(),
+          new SetAcl()
+      ).build();
+
+  @Parameter(required = true, names = {"-o", "--outputDirectory"},
+      description = "The directory to write generated journal artifacts to")
+  private String mOutputDirectory;
+
+  /**
+   * @return the output directory
+   */
+  public String getOutputDirectory() {
+    return mOutputDirectory;
+  }
+
+  /**
+   * Generates journal files to be used by the backwards compatibility test. The files are named
+   * based on the current version defined in ProjectConstants.VERSION. Run this with each release,
+   * and commit the created journal and snapshot into the git repository.
+   *
+   * @param args no args expected
+   */
+  public static void main(String[] args) throws Exception {
+    BackwardsCompatibilityJournalGenerator generator = new BackwardsCompatibilityJournalGenerator();
+    new JCommander(generator, args);
+    if (!LoginUser.get().getName().equals("root")) {
+      System.err
+          .printf("Journals must be generated as root so that they can be replayed by root%n");
+      System.exit(-1);
```

Comment:
Is there no better way, ie. ensure the permission is acceptable?

Reply:
With security enabled, the username of the user that restarts Alluxio has to match exactly with the user that originally started Alluxio: https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java#L601